src/scheduler.ml

```diff
@@ -140,125 +218,147 @@ end = struct
 
     let rec buffer_thread () =
```

Comment:
Could you add a comment for why we need a separate thread to buffer, rather than just firing the events

Reply:
Done